railgun/test/integration/suite_test.go

```diff
@@ -44,6 +44,9 @@ const (
 	// httpcTimeout is the default client timeout for HTTP clients used in tests.
 	httpcTimeout = time.Second * 3
 
+	// ingressClass indicates the "kubernetes.io/ingress.class" which the tests will use for supported object reconcilation
```

Comment:
I think that we should test all combinations of ingress class settings (for the ingressv1 type) in the integration test.
This means at least:
- `Ingress` with no annotation and with no `IngressClassName`
- `Ingress` with an annotation `not-kongtests`
- `Ingress` with an annotation `kongtests`
- `Ingress` with a `spec.IngressClassName` `not-kongtests`
- `Ingress` with a `spec.IngressClassName` `kongtests`

Reply:
Given that none of the controllers meaningfully manager this field yet and writing the tests would either increase scope or be ultimately contrived I've created a follow up with these as AC: https://github.com/Kong/kubernetes-ingress-controller/issues/1266

LMKWYT